lib/chef/provider/package/debian.rb

```diff
@@ -0,0 +1,135 @@
+#
+# Author:: Kapil Chouhan (<kapil.chouhan@msystechnologies.com>)
+# Copyright:: Copyright 2008-2018, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/provider/package"
+
+class Chef
+  class Provider
+    class Package
+      class Debian < Chef::Provider::Package
+        use_multipackage_api
+
+        action :install do
```

Comment:
this is all a bit problematic because now we've forked the install action in the subclass, so fixes to the superclass would need to get replicated into this code.

what likely needs to get extracted is a hook in the superclass like `def prepare_for_installation` which is called from the middle of the `action :install` in the superclass but which is normally a method with no code, then the subclass can override the implementation of that to do the response_file/preseed_file stuff.   then there is no copypasta between the superclass and the subclass.